### PR TITLE
issue #8: Adding new workflow

### PR DESCRIPTION
**Pull request by SamuelPelletierEvraire**
## Tasks
- [X] Research and choose a suitable workflow model (e.g., Gitflow, GitHub Flow).
- [X] Define the branching strategy based on the selected workflow model.
- [X] Set up branch protection rules to enforce the chosen workflow.
- [X] Configure automated testing and code review processes within the workflow.

## Acceptance Criteria
- The chosen workflow model is documented and clearly communicated to all team members.
- Branching strategy is established according to the selected workflow model.
- Branch protection rules are applied to enforce the workflow guidelines.
- Automated testing and code review processes are integrated into the workflow.

### CODE REVIEW

**Comment by SamuelPelletierEvraire** [comment]
For now i'm trying to fix all the issue that i have on the code and after i'm gonna make the documentation

**Comment by rngadam** [comment]
> For now i'm trying to fix all the issue that i have on the code and after i'm gonna make the documentation

so why is there a checkmark on the task?

**Comment by rngadam** [comment]
> For now i'm trying to fix all the issue that i have on the code and after i'm gonna make the documentation

this is not the right approach ; you have a PR here that is about "adding a new workflow" so you should ensure that it is reviewed and merged. if you want to keep making changes that are unrelated to "adding a new workflow" you should branch off this one and work on that separetely so reviewers don't have to chase a constantly changing PR.

**Comment by SamuelPelletierEvraire** [comment]
@rngadam  do you know why lint-test always fail I don't find why.

**Comment by rngadam** [comment]
image

it's trying to import tests but you've named your folder Tests with uppercase T. this is why we have a naming convention to have all names lowercases. your tests themselves don't follow the naming convention. you've also checked in __pycache__ and compiled versions of the python script.

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,4 @@ keys.json
 questions.json
 keys_questions.txt
 questions_spreadsheet.csv
-base_composition_questions.csv
\ No newline at end of file
+base_composition_questions.csv
```
Comment (MaxenceGui) [inline .gitignore:5]: add *.pyc to ignore __pycache__ folder

**Review by MaxenceGui — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -1,2 +1,3 @@
 # harvester-prototype
-This repository is for the frontend of the project react 
+
+This repository is for the frontend of the project react
```
Comment (rngadam) [inline README.md:3]: description incorrect

**Review by MaxenceGui — APPROVED** [review]

**Comment by rngadam** [comment]
I don't think this should have been merged. it was far from done. it should have been rebased on main at least as the pyc and pycache are now checked in the history and the commit history looks really noisy:

image

**Pull request merged**